### PR TITLE
[RPA-839] User deletion in not activated users management page

### PR DESCRIPTION
**Pull request by Aeri4a**
## Description
Delete confirm modal - waiting for action
sc1

Delete confirm modal - loading
sc2

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts [hunk 1]
```diff
@@ -48,7 +48,18 @@ const buildUsersExtraReducers = (builder: ActionReducerMapBuilder<UsersState>) =
             state.notActivated.loading = false;
         })
         .addCase(updateNotActivated.rejected, (state) => {
-            state.notActivated.loading = false;
+            state.loading = false;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts]: i belive that's unintended

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts [hunk 2]
```diff
@@ -48,7 +48,18 @@ const buildUsersExtraReducers = (builder: ActionReducerMapBuilder<UsersState>) =
...
+
+        // DELETE USER
+        .addCase(deleteUser.pending, (state) => {
+            state.loading = true;
+        })
+        .addCase(deleteUser.fulfilled, (state) => {
+            state.loading = false;
+        })
+        .addCase(deleteUser.rejected, (state) => {
+            state.loading = false;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts]: be more specific like with state.notActivated.loading

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/index.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import modalsActionsTranslations from './modals';
+
+const usersActionsTranslations = {
+    ...modalsActionsTranslations,
+};
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/index.ts:5]: will be more there?

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import tableRegisterTranslations from './table';
+import viewRegisterTranslations from './view';
+
+const usersViewTranslations = {
+    ...viewRegisterTranslations,
+    ...tableRegisterTranslations,
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/index.ts]: this register is stabbing my soul.
I suggest to take a while to rename all registry files and function names to registration - now you do have far fewer changes and up-to-date branch

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/index.ts]: F2

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/table.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const tableRegisterTranslations = {
+    'Users.Register.Table.Columns.Email': 'Email',
+    'Users.Register.Table.Columns.CreateDate': 'Create date',
+    'Users.Register.Table.Columns.Role': 'Role',
+    'Users.Register.Table.Error.Rows': 'No requests found'
+};
+
+export default tableRegisterTranslations;
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/table.ts]: does your ESLint work? Should throw error "newline required at the end of file"

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import englishViewRegisterTranslations from '#src-app/translations/en/users/register/view';
+
+const viewRegisterTranslations: typeof englishViewRegisterTranslations = {
+    'Users.Register.View.Events.Success.AcceptingUser': 'Sukces',
+    'Users.Register.View.Events.Error.RolesNotSelected': 'Nie wybrano wszystkich wymaganych roli',
+    'Users.Register.View.Events.Error.AcceptFailed': 'Nie można zaakceptować wybranych użytkowników',
+    'Users.Register.View.Events.Error.FindingUsers': 'Wystąpił błąd podczas znajdowania nie aktywowanych użytkowników',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts]: nieaktywowanych razem

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const viewRegisterTranslations = {
+    'Users.Register.View.Events.Success.AcceptingUser': 'Success',
+    'Users.Register.View.Events.Error.RolesNotSelected': 'Not selected all required roles',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts]: rename to: Not all users have been assigned roles

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+const viewRegisterTranslations = {
+    'Users.Register.View.Events.Success.AcceptingUser': 'Success',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts]: Successfully accepted users

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    'Users.Register.View.Events.Error.RolesNotSelected': 'Not selected all required roles',
+    'Users.Register.View.Events.Error.AcceptFailed': 'Could not accept selected users',
+    'Users.Register.View.Events.Error.FindingUsers': 'Error finding not activated users',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/view.ts]: is it ERROR (e.g. we get 404/403/501) or is it MESSAGE (that we search but, there are no users in DB)?

if it's a MESSAGE rename translate key and translation to "Could not find activated users"

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/modals.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
+const modalsActionsTranslations = {
+    'Users.Actions.Modals.DeleteModal.Title': 'Delete users',
+    'Users.Actions.Modals.DeleteModal.Success': 'Successfully deleted selected users',
+    'Users.Actions.Modals.DeleteModal.Error': 'Error while deleting users',
+    'Users.Actions.Modals.DeleteModal.Content': 'Are you sure you want to delete following users?',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/modals.ts]: Are you sure you want to delete the following users?

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import englishViewRegisterTranslations from '#src-app/translations/en/users/register/view';
+
+const viewRegisterTranslations: typeof englishViewRegisterTranslations = {
+    'Users.Register.View.Events.Success.AcceptingUser': 'Sukces',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts]: adjust according to en translations

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+    'Users.Register.View.Events.Error.RolesNotSelected': 'Nie wybrano wszystkich wymaganych roli',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts]: Nie wszystkim użytkownikom nadano role

Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts]: czyli w domyśle (przydzielenie roli każdemu użytkownikowi jest wymagane)

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    'Users.Register.View.Events.Error.AcceptFailed': 'Nie można zaakceptować wybranych użytkowników',
+    'Users.Register.View.Events.Error.FindingUsers': 'Wystąpił błąd podczas znajdowania nie aktywowanych użytkowników',
+    'Users.Register.View.SearchBarPlaceholder': 'Wyszukaj',
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/translations/pl/users/register/view.ts]: on the blog there's "szukaj" 
image

stick to it

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.styles.ts [hunk 14]
```diff
@@ -40,7 +40,7 @@ export const DeleteButton = styled(Button)(({theme}) => `
         && {
             margin-left: 20px;
         }
-        
+
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.styles.ts:43]: the best extensions - changes nothing to more nothing
RaingumpKtoGIF

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx [hunk 15]
```diff
@@ -42,6 +43,7 @@ const UsersRegisterView: FC = () => {
     const { notActivated } = useSelector(usersSelector);
     const { search, handleSearch, refreshSearch } = useUserSearch(false, limit, page);
 
+    const [showDeleteDialog, setShowDeleteDialog] = useState(false);
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx]: isDeleteDialogVisible or isDeleteDialogShown

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx [hunk 16]
```diff
@@ -105,6 +107,11 @@ const UsersRegisterView: FC = () => {
 
     return (
         <>
+            <DeleteUserDialog
+                open={showDeleteDialog}
+                onClose={() => setShowDeleteDialog(false)}
+                getSelectedUsers={sendSelectedUsersToDeleteModal}
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx]: avoid linking words - here "to"

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx [hunk 17]
```diff
@@ -84,9 +86,9 @@ const UsersRegisterView: FC = () => {
                 enqueueSnackbar(translate('Users.Register.View.Events.Error.AcceptFailed'), { variant: 'error' });
             });
 
-    const handleDelete = () => {
-        // to be implemented soon
-    };
+    const handleDelete = () => setShowDeleteDialog(true);
+
+    const sendSelectedUsersToDeleteModal = (): IUser[] => notActivated.allByPage.content.filter((user) => selections.includes(user.id));
```
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/UsersRegisterView/UsersRegisterView.tsx]: you do use it to pass user's to delete modal but there's nothing about the modal in function body.

rename to sth like getSelectedUsers

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/users/DeleteUser.tsx
Comment (star-in-the-jar) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/users/DeleteUser.tsx:1]: rename file to DeleteUserDialog

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]

**Comment by star-in-the-jar** [comment]
bump version

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts (hunk 2 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts]: That function will be for both activated and not activated so I thought about using more general param. But if there is better solution then I am open to suggestions.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/index.ts (hunk 3 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/actions/index.ts:5]: I think so. Actions like editing user should be also placed in here in further users features.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/index.ts (hunk 4 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/index.ts]: OK.

File: runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/table.ts (hunk 5 above)
Comment (Aeri4a) [reply runbotics/runbotics-orchestrator-ui/src/src-app/translations/en/users/register/table.ts]: There is actually a newline. 🤔

File: runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts (hunk 2 above)
Comment (star-in-the-jar) [reply runbotics/runbotics-orchestrator-ui/src/src-app/store/slices/Users/Users.extraReducers.ts]: as we discussed -> this + loading circle on delete users dialog
-> backend merge + rebase to develop + test this config -> ping me/RPACR on teams

**Review by star-in-the-jar — CHANGES_REQUESTED** [review]

**Review by star-in-the-jar — APPROVED** [review]

**Pull request merged**